src/httpserver.cpp

```diff
@@ -646,6 +670,11 @@ HTTPRequest::RequestMethod HTTPRequest::GetRequestMethod()
     }
 }
 
+bool EqualHeaders(const std::string& h1, const std::string& h2)
+{
+    return evutil_ascii_strcasecmp(h1.c_str(), h2.c_str()) == 0;
```

Comment:
Is there a reason for relying on `event2` in this function? If not I suggest doing this without relying on third-party code by introducing the required locale independent string helper functions in-tree. That would perhaps also allow us to reduce the number of know locale dependence problems listed here: https://github.com/bitcoin/bitcoin/blob/master/test/lint/lint-locale-dependence.sh#L4